internal/bminventory/inventory.go

```diff
@@ -1973,10 +1959,28 @@ func (b *bareMetalInventory) updateClusterData(ctx context.Context, cluster *com
 		}
 	}
 
+	if params.ClusterUpdateParams.MachineNetworkCidr != nil && common.IsSingleNodeCluster(cluster) {
+		machineCidr = swag.StringValue(params.ClusterUpdateParams.MachineNetworkCidr)
+		if err = network.VerifySubnetCIDR(machineCidr); err != nil {
+			log.WithError(err).Warningf("Given machine cidr %q is not valid", machineCidr)
+			return common.NewApiError(http.StatusBadRequest, err)
+		}
+		err = setCommonUserNetworkManagedParams(params.ClusterUpdateParams, common.IsSingleNodeCluster(cluster), machineCidr, updates, log)
+		if err != nil {
+			return err
+		}
+
+		setMachineNetworkCIDRForUpdate(updates, machineCidr)
```

Comment:
This gets called in `setCommonUserNetworkManagedParams` don't think you need it here as well

Reply:
yeah forgot to remove it from here

Reply:
thanks